opm/grid/common/MetisPartition.cpp

```diff
@@ -0,0 +1,282 @@
+/*
+  Copyright 2024 OPM-OP AS
+
+  This file is part of The Open Porous Media project  (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#if HAVE_MPI // no code in this file without MPI, then skip includes.
+#include <opm/common/OpmLog/OpmLog.hpp>
+#include <opm/grid/common/ZoltanGraphFunctions.cpp>
+#include <opm/grid/common/MetisPartition.hpp>
+#include <opm/grid/utility/OpmWellType.hpp>
+#include <opm/grid/cpgrid/CpGridData.hpp>
+#include <opm/grid/cpgrid/Entity.hpp>
+#include <algorithm>
+#include <type_traits>
+#endif
+
+#if defined(HAVE_METIS) && HAVE_MPI
+namespace Dune
+{
+namespace cpgrid
+{
+
+void setMetisOptions(const std::map<std::string, std::string>& optionsMap, idx_t* options) {
+    // Initialize all options to default values
```

Comment:
This function is partly responsible for the jenkins failure. Scotch does not support the defines.

Maybe we have a surrounding
```
#ifndef HAVE_PTSCOTCH

#endif 
```
in this function